### PR TITLE
Cleanup `WbNetwork` to a singleton class

### PR DESCRIPTION
**Pull request by ygoumaz**
The way `WbNetwork` is used is not consistent. Sometimes it is called as a singleton, sometimes not. This PR cleans up the class to get consistent use as a singleton class. 

The side effect of this inconsistency is that the value of `gCacheMap` was not the same in all calls to the `get`, `isCached` and `getUrlFromEphemeralCache` methods. This is leading to errors when loading worlds that have some PROTOs already present in the assets cache, e.g. the worlds loaded in the `world update` test of the CI (see #5087).

### CODE REVIEW

File: src/webots/core/WbNetwork.cpp [hunk 1]
```diff
@@ -44,7 +40,7 @@ WbNetwork *WbNetwork::instance() {
 
 WbNetwork::WbNetwork() {
   mNetworkAccessManager = NULL;
-  gCacheMap.clear();
+  mCacheMap.clear();
```
Comment (omichel) [inline src/webots/core/WbNetwork.cpp:43]: This is not needed as a `QMap` is initially empty.
```suggestion
```

File: src/webots/core/WbNetwork.cpp [hunk 2]
```diff
@@ -118,28 +114,28 @@ void WbNetwork::save(const QString &url, const QByteArray &content) {
...
       // save reference in internal representation
-      gCacheMap.insert(url, path);
+      mCacheMap.insert(url, path);
     }
   }
 }
 
-const QString &WbNetwork::get(const QString &url) {
-  if (!gCacheMap.contains(url)) {
+const QString WbNetwork::get(const QString &url) {
```
Comment (omichel) [inline src/webots/core/WbNetwork.cpp:122]: Why did you remove the reference here?

**Review by omichel — CHANGES_REQUESTED** [review]

File: src/webots/core/WbNetwork.cpp (hunk 2 above)
Comment (ygoumaz) [reply src/webots/core/WbNetwork.cpp:121]: This reference was not present before the changes applied by @stefaniapedrazzi in #5023. So I thought it was not useful. If we use a reference for the `get` method, should we also use one for `getUrlFromEphemeralCache`?

Comment (omichel) [reply src/webots/core/WbNetwork.cpp:121]: Yes, that's probably better.

Comment (ygoumaz) [reply src/webots/core/WbNetwork.cpp]: Actually, `getUrlFromEphemeralCache` is returning a temporary object, so it is not a good idea to use a reference. I still put it back for the `get` method.

Comment (stefaniapedrazzi) [reply src/webots/core/WbNetwork.cpp]: We should always return the reference instead of a copy when possible.

**Comment by omichel** [comment]
The CI test failed because:
```
src/webots/nodes/WbCadShape.cpp:233:39: warning: Assert statement calls a function which may have desired side effects: 'isCached'. [assertWithSideEffect]
        assert(WbNetwork::instance()->isCached(adjustedUrl));
```
I believe we wrote this assert mainly because the `isCached()` method name is misleading: this is not a const method as we would expect it with such a name. It does possibly modify the object, which is not expected, hence the warning... So, I believe we should rename this method and probably remove the assert or use a different one.

**Comment by ygoumaz** [comment]
All tests passed, so this PR is ready for review.

File: src/webots/core/WbNetwork.hpp [hunk 3]
```diff
@@ -25,15 +25,20 @@ class WbNetwork {
   QNetworkAccessManager *networkAccessManager();
   void setProxy();
 
-  static bool isCached(const QString &url);
-  static const QString &get(const QString &url);
+  // check the file in the cache and add it to the internal representation. Outside of the assets, this function should be used
+  // above isCachedNoMapUpdate().
+  bool isCachedWithMapUpdate(const QString &url);
+  // check the file in the cache without adding it to the internal representation. This one should only be used in assertions
+  // because they cannot have side effects.
```
Comment (omichel) [inline src/webots/core/WbNetwork.hpp:32]:
```suggestion
  // check the file in the cache without adding it to the internal representation (for use in asserts, no side effect).
```

File: src/webots/core/WbNetwork.hpp [hunk 4]
```diff
@@ -25,15 +25,20 @@ class WbNetwork {
   QNetworkAccessManager *networkAccessManager();
   void setProxy();
 
-  static bool isCached(const QString &url);
-  static const QString &get(const QString &url);
+  // check the file in the cache and add it to the internal representation. Outside of the assets, this function should be used
+  // above isCachedNoMapUpdate().
```
Comment (omichel) [inline src/webots/core/WbNetwork.hpp:29]:
```suggestion
  // check the file in the cache and add it to the internal representation (don't use it in asserts).
```

**Review by omichel — CHANGES_REQUESTED** [review]
Thank you, it looks good to me.
I suggest to slightly improve the comments.

**Review by omichel — APPROVED** [review]
Thank you.
